addons/account/models/sequence_mixin.py

```diff
@@ -253,3 +256,29 @@ def _is_last_from_seq_chain(self):
         seq_format, seq_format_values = self._get_sequence_format_param(last_sequence)
         seq_format_values['seq'] += 1
         return seq_format.format(**seq_format_values) == self.name
+
+    def _is_end_of_seq_chain(self):
+        """Tells whether or not these elements are the last ones of the sequence chain.
+
+        :return: True if self are the last elements of the chain.
+        """
+        sorted = defaultdict(lambda: {'record': self.browse(), 'seq_list': []})
```

Comment:
Do you need that 'record' key ? Why ?

Also, you're shadowing sorted() here :p